### ISSUE TITLE
Spears (including the gungir) should be throwable, otherwise too OP

### ISSUE THREAD

**Issue by KaylebJay** [KaylebJay/tunnelers-bugz#83]
Make the spears throwable: left-click, throws according to the player's yaw perhaps. 
Then, if it hits a mob/entity it damages them by the given amounts.
This would dramatically change the way-too-OP spears like the Gungir and make spears more like arrows for a quick damaging and then take it up with a sword.

**Comment by KaylebJay** [comment]
With this: dragons would no longer damage the max damage, probably more like 5 HP each fireball.

**Comment by fluxionary** [comment]
the really OP thing about the gungir is that its range is way farther than the dragons "punch" (and the gungir has an absurd knockback). if you made it an obligatorily projectile weapon, i'd just use a multitool to keep the dragons at a distance. another possible way to fix the issues would be if the dragons breath attacks were fixed and the spears' ranges were reduced some. 

the spears also have an incredibly useful second ability: they can trim leaves (or fruit) off of the top of trees, and harvest crops very efficiently, which would be ruined by making them projectile-only...

**Comment by KaylebJay** [comment]
That's true, its super OP in that regard. plus 165DMG is just way too much.
You could make the spear ranges reduced, yes... not sure though. Throwing still seems to be the best solution.

Swords should be able to do that just as well, and if you reduced the range it would make them just like swords.

**Comment by fluxionary** [comment]
in my mind, a reduced-range spear would still have significantly more range than a sword. currently the gungir is one of my top-3 used tools; not just because it kills things, but for that reach. i also use it to poke distant matrix walls in caves to force them to load.

**Comment by KaylebJay** [comment]
Well, it could be like left-click does just what it always does, but it doesn't damage mobs, but right-click will throw it as a projectile. 
This is controversial I see, so please give me alternative ideas.

**Comment by fluxionary** [comment]
i thought about the "left click/right click" thing too. perhaps if left click (stab) doesn't cause much damage but has the other effects (like cutting leaves/crops); if you want to cause a lot of damage, you have to throw it (right click).

**Comment by KaylebJay** [comment]
Right, that's my current thought.. hm. 
Dragons could also drop eggs more frequently once again if this happens because they won't be a piece of cake to defeat :)

**Comment by KaylebJay** [comment]
My current problem is the damaging system of petz. So far they can be knocked back by spears but do not get damaged.

**Comment by KaylebJay** [comment]
Pretty close to committing, so  will close when committed. Making them throwable (I was able to get it working) is not a good idea. In cases of lag the entity can teleport or even dissapear/glitch into nodes, and considering the gungir and other spears are so valuable, I will not do this. Instead, the full punch interval has been made much slower, the range decreased from 12 to 7, and a few other various 'improvements'. This will meant that petz are still instantly 'killable' with the gungir, but dragons and the new paleotest mobs will be a real challenge. I did find a way to kill a Tyrannosaurus  pretty quickly, but it requires some skill :)

In the future, what I plan to happen is the spear damages more, or less, depending on how close you hit to the center of the mob's hitbox, and then we could unnerf the full_punch_interval. For now, the current implementation should be fine though.
